File: src/main/java/kdh/boardproject/dto/board/BoardListDto.java
package kdh.boardproject.dto.board;

import kdh.boardproject.entity.Board;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class BoardListDto {
    private Long idx;
    private String title;
    private String content;
    private LocalDateTime createdAt;

    public BoardListDto(Board board){
        this.idx = board.getIdx();
        this.title = board.getTitle();
        this.content = board.getContent();
        this.createdAt = board.getCreatedAt();
    }
}

File: src/main/java/kdh/boardproject/dto/board/BoardListDto.kt
package kdh.boardproject.dto.board

import kdh.boardproject.entity.Board
import java.time.LocalDateTime


class BoardListDto(board: Board) {
     val idx: Long?
     val title: String
     val content: String
     val createdAt: LocalDateTime?

    init {
        idx = board.idx
        title = board.title
        content = board.content
        createdAt = board.createdAt
    }
}